### PR TITLE
Delegate global definitions to the global autoloader

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

All this code is executed during the bootstraping of GLPI, whatever the context is (UI, console, cron), and can then be executed automatically using the files autoloading feature of composer. The execution order of the moved logic does not matter.

Here is the detailled list of changes:
- Autoload the `DBUtils` function aliases
- Autoload the legacy autoloader
- Autoload the i18n functions
- Autoload the `isPluginItemtype()` function
- Autoload the `isApi()` function
- Autoload the `isCommandLine()` function
- Delegate the non configurable constants definition
- Delegate the `$CFG_GLPI` global variable initialization

### CODE REVIEW

File: src/_/CFG_GLPI.php [hunk 1]
```diff
@@ -643,49 +553,49 @@
...
+    Notification::class       => "pics/impact/notification.png",
+    Profile::class            => "pics/impact/profile.png",
+    Project::class            => "pics/impact/project.png",
+    Rack::class               => "pics/impact/rack.png",
+    SLM::class                => "pics/impact/slm.png",
+    SoftwareLicense::class    => "pics/impact/softwarelicense.png",
+    Supplier::class           => "pics/impact/supplier.png",
+    User::class               => "pics/impact/user.png",
+    Database::class           => "pics/impact/database.png",
 ];
```
Comment (cedric-anne) [inline src/_/CFG_GLPI.php:599]: Using `::class` here result in the fact that all these classes are not loaded anymore during the GLPI bootstraping. I do not know if it improves the bootstraping duration (maybe a few milliseconds, but nor noticeable when opcache is already populated).

The changes made in tests were required due to this change.

**Comment by Pierstoval** [comment]
> I do not reveiw everything, but I'm not really fan of the `_` directory name.

Same here. Since it's all related to autoload, naming the directory `src/autoload/` would be more logical

**Comment by cedric-anne** [comment]
> I do not reveiw everything, but I'm not really fan of the `_` directory name.

As it contains functions, constants and global declarations, I was not able to find a relevant name. It can be changed easilly.

> Also, any config is not stricto sensu part of source; I guess we should keep only Classes that are meant to be autoloaded in `src` directory - and nothing else.

The long term plan is to replace most of this code by classes handled directly by a dependency injection system (that will be added by #17213 ). This PR was made to simplify the transition. I only moved, in these autoloaded files, pieces of code that corresponds to pure hardcoded declarations. It is a small step to simplify the `includes` logic, in order to make the job started in #17278 a bit simplier.

**Comment by cedric-anne** [comment]
> > I do not reveiw everything, but I'm not really fan of the `_` directory name.
> 
> Same here. Since it's all related to autoload, naming the directory `src/autoload/` would be more logical

Indeed, naming it `src/autoload` would be more logical.

**Comment by cedric-anne** [comment]
> > > I do not reveiw everything, but I'm not really fan of the `_` directory name.
> > 
> > 
> > Same here. Since it's all related to autoload, naming the directory `src/autoload/` would be more logical
> 
> Indeed, naming it `src/autoload` would be more logical.

I renamed the directory.

**Review by orthagh — APPROVED** [review]

**Pull request merged**